### ISSUE TITLE
Decky tailscale controller broke tailscaled 

### ISSUE THREAD

**Issue by Angel4371** [tailscale-dev/deck-tailscale#31]
Been trying to fix this for hours 

Ran the uninstall and re install script 

Tried to get the qr code 

Says to run sudo systemctl start tailscaled

Get this error 

Job for tailscaled.service failed because the control process exited with error code.
See "systemctl status tailscaled.service" and "journalctl -xeu tailscaled.service" for details.

Try to get info and it returns with this 

tailscaled.service - Tailscale node agent
     Loaded: loaded (/etc/systemd/system/tailscaled.service; disabled; preset: disabled)
     Active: failed (Result: exit-code) since Fri 2024-12-13 02:42:15 CST; 6min ago
       Docs: https://tailscale.com/kb/
    Process: 6948 ExecStart=/usr/sbin/tailscaled --state=/var/lib/tailscale/tailscaled.state --socket=/run/tailscale/tailscaled.sock --port=${PORT} $FLAGS >
    Process: 6950 ExecStopPost=/usr/sbin/tailscaled --cleanup (code=exited, status=203/EXEC)
   Main PID: 6948 (code=exited, status=203/EXEC)
        CPU: 11ms

Dec 13 02:42:15 (nsfw system name) systemd[1]: tailscaled.service: Scheduled restart job, restart counter is at 5.
Dec 13 02:42:15 (nsfw system name) systemd[1]: tailscaled.service: Start request repeated too quickly.
Dec 13 02:42:15 (nsfw system name) systemd[1]: tailscaled.service: Failed with result 'exit-code'.
Dec 13 02:42:16 (nsfw system name)systemd[1]: Failed to start Tailscale node agent.
Lines 1-13/13 (END)

I have no idea what to do, please help.

**Comment by Angel4371** [comment]
Will do very soon, thank you for the quick reply

**Comment by Angel4371** [comment]
░░ The process' exit code is 'exited' and its exit status is 203.
Dec 14 22:49:43 (system name) systemd[1]: tailscaled.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ The unit tailscaled.service has entered the 'failed' state with result 'exit-code'.
Dec 14 22:49:43 (system name) systemd[1]: Failed to start Tailscale node agent.
░░ Subject: A start job for unit tailscaled.service has failed
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit tailscaled.service has finished with a failure.
░░ 
░░ The job identifier is 4170 and the job result is failed.
Dec 14 22:49:43 (system name)[1]: tailscaled.service: Scheduled restart job, restart counter is at 5.
░░ Subject: Automatic restarting of a unit has been scheduled
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ Automatic restarting of the unit tailscaled.service has been scheduled, as the result for
░░ the configured Restart= setting for the unit.
Dec 14 22:49:43 (system name)[1]: tailscaled.service: Start request repeated too quickly.
Dec 14 22:49:43 (system name) systemd[1]: tailscaled.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ The unit tailscaled.service has entered the 'failed' state with result 'exit-code'.
Dec 14 22:49:43 (system name) systemd[1]: Failed to start Tailscale node agent.
░░ Subject: A start job for unit tailscaled.service has failed
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit tailscaled.service has finished with a failure.
░░ 
░░ The job identifier is 4298 and the job result is failed.

**Comment by legowerewolf** [comment]
203, hm? Are you sure the Tailscale binary's in the right place and is executable?

**Comment by Angel4371** [comment]
How do I make sure of this? Sorry, I am a bit inexperienced with this

**Comment by legowerewolf** [comment]
What do you get when you run `ls -la /opt/tailscale` and `cat /etc/systemd/system/tailscaled.service`?

**Comment by Angel4371** [comment]
total 54556
drwxr-xr-x 2 root root     4096 Sep  7 17:56 .
drwxr-xr-x 3 root root     4096 Sep  7 17:56 ..
-rwxr-xr-x 1 root root 19309431 Dec 13 01:34 tailscale
-rwxr-xr-x 1 root root 36539989 Dec 13 01:34 tailscaled


[Unit]
Description=Tailscale node agent
Documentation=https://tailscale.com/kb/
Wants=network-pre.target
After=network-pre.target NetworkManager.service systemd-resolved.service

[Service]
EnvironmentFile=/etc/default/tailscaled
ExecStart=/usr/sbin/tailscaled --state=/var/lib/tailscale/tailscaled.state --socket=/run/tailscale/tailscaled.sock --port=${PORT} $FLAGS
ExecStopPost=/usr/sbin/tailscaled --cleanup

Restart=on-failure

RuntimeDirectory=tailscale
RuntimeDirectoryMode=0755
StateDirectory=tailscale
StateDirectoryMode=0700
CacheDirectory=tailscale
CacheDirectoryMode=0750
Type=notify

[Install]
WantedBy=multi-user.target

**Comment by legowerewolf** [comment]
Hm. So it looks like your Tailscale binaries are good. With _this_ config file, they're in the wrong place, but... I didn't think to ask before, but is there a systemd override file? Is there a directory `/etc/systemd/system/tailscaled.service.d/` and what, if anything, is in there?

**Comment by Angel4371** [comment]
I have something called override.conf.bak

**Comment by legowerewolf** [comment]
And that's the only file in there? Seems like you need to download override.conf from this repo and put it in that directory.

**Comment by Angel4371** [comment]
it's not letting me put anything in that folder, is there something special I have to do to let me put it in there?

**Comment by legowerewolf** [comment]
Something like `sudo mv path/to/downloaded/override.conf /etc/systemd/system/tailscaled.service.d/override.conf` ought to work.

**Comment by Angel4371** [comment]
yeah just changed the ownership of the folder to deck which let me put it in there, with the override.conf and override.conf.bak in there is there anything else that should be in there?

**Comment by Angel4371** [comment]
seems like that didn't fix it, I'm still not able to connect to my home pc while out on my hotspot

**Comment by legowerewolf** [comment]
`sudo systemctl daemon-reload` to make sure its loaded the changes

**Comment by Angel4371** [comment]
it's still saying that all devices are unavailable to connect to

**Comment by Angel4371** [comment]
do I have to return ownership of the folder to root?

**Comment by legowerewolf** [comment]
ccing @diericx because it's his install method

**Comment by Angel4371** [comment]
I'm also attempting to run sudo bash tailscale.sh to see if I need to update it and it's saying "no such file or directory"

**Comment by Angel4371** [comment]
> ccing @diericx because it's his install method

alright, thank you :)

**Comment by Angel4371** [comment]
the issue of it connecting has been solved, but now on reboot it doesn't auto connect to the tailscale server, what should I run to make sure it runs on startup now?

**Comment by legowerewolf** [comment]
`sudo systemctl enable tailscaled --now`

**Comment by Erisa** [comment]
Going to close this since it seems answered **but** if i'm wrong to do that please let me know!
